lib/base/view/base_view.dart

```diff
@@ -0,0 +1,231 @@
+import 'package:flutter/material.dart';
+import 'package:githubdummy/base/view_model/base_view_model.dart';
+import 'package:githubdummy/widgets/app_bar_view.dart';
+import 'package:githubdummy/widgets/loading_view.dart';
+import 'package:go_router/go_router.dart';
+import 'package:provider/provider.dart';
+
+import '../../config/themes/theme.dart';
+
+abstract class BaseView<T extends BaseViewModel> extends StatelessWidget {
+  final T viewModel;
+  final Widget? child;
+
+  const BaseView({
+    super.key,
+    required this.viewModel,
+    this.child,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider<T>(
+      create: (context) => viewModel,
+      child: Consumer<T>(
+        builder: _buildScreen,
+      ),
+    );
+  }
+
+  Widget _buildScreen(BuildContext context, T viewModel, Widget? child) {
+    if (viewModel.errorMessage != null) {
+      return _buildError(context, viewModel.errorMessage!);
+    }
```

Comment:
I think we can just show error message using snackbar or toast. Don't need to replace the main screen with a error screen.

Reply:
Alright. Changed it.